runtime/frequency/src/lib.rs

```diff
@@ -946,6 +946,10 @@ impl_runtime_apis! {
 				Err(_) => None,
 			}
 		}
+
+		fn get_msa_keys_offchain(msa_id: MessageSourceId) -> (BlockNumber, Vec<AccountId>) {
+			Msa::get_msa_keys_offchain(msa_id)
+		}
```

Comment:
Called by RPC , return type is a tuple (BlockNumber, Vec<AccontId>)
BlockNumber states the latest snapshot of when offchain index was updated